src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs

```diff
@@ -0,0 +1,488 @@
+                        editSession.TrackDocumentWithReportedDiagnostics(document.Id);
+
+                        // Synthesize external location for the diagnostic.
+                        // Currently we need to do that bacause 
+                        // https://github.com/dotnet/roslyn/issues/34360
+                        return diagnostics.SelectAsArray((d, l) => d.ToDiagnostic(l), Location.Create(document.FilePath, textSpan: default, lineSpan: default));
+                    }
+
+                    // Check if EnC is allowed in the current state of the debugee.
+                    var debuggeStateDiagnostics = editSession.GetDebugeeStateDiagnostics();
+                    if (!debuggeStateDiagnostics.IsEmpty)
+                    {
+                        return reportDiagnostics(debuggeStateDiagnostics);
+                    }
+
+                    // Check if EnC is allowed for all loaded modules corresponding to the project.
+                    var moduleDiagnostics = editSession.GetModuleDiagnostics(mvid, project.Name);
+                    if (!moduleDiagnostics.IsEmpty)
+                    {
+                        return reportDiagnostics(moduleDiagnostics);
+                    }
+                }
+
+                if (analysis.RudeEditErrors.IsEmpty)
+                {
+                    return ImmutableArray<Diagnostic>.Empty;
+                }
+
+                editSession.Telemetry.LogRudeEditDiagnostics(analysis.RudeEditErrors);
+
+                // track the document, so that we can refresh or clean diagnostics at the end of edit session:
+                editSession.TrackDocumentWithReportedDiagnostics(document.Id);
+
+                var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                return analysis.RudeEditErrors.SelectAsArray((e, t) => e.ToDiagnostic(t), tree);
+            }
+            catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
+            {
+                return ImmutableArray<Diagnostic>.Empty;
+            }
+        }
+
+        private ImmutableArray<Diagnostic> GetRunModeDocumentDiagnostics(Document document)
+        {
+            lock (_documentsWithReportedDiagnosticsDuringRunModeGuard)
+            {
+                _documentsWithReportedDiagnosticsDuringRunMode.Add(document.Id);
+            }
+
+            // Synthesize external location for the diagnostic.
+            // Currently we need to do that bacause of https://github.com/dotnet/roslyn/issues/34360
+            var location = Location.Create(document.FilePath, textSpan: default, lineSpan: default);
+            var descriptor = EditAndContinueDiagnosticDescriptors.GetDescriptor(EditAndContinueErrorCode.ChangesNotAppliedWhileRunning);
+
+            return ImmutableArray.Create(Diagnostic.Create(descriptor, location, new[] { document.Project.Name }));
+        }
+
+        private void ClearReportedRunModeDiagnostics()
```

Comment:
```suggestion
        private void ReanalyzeDocumentsWithReportedRunModeDiagnostics()
```

Reply:
The intent of the method is to clear the diagnostics reported during run mode. This is done by invoking Reanalyze but that's an implementation detail of the method.

---
In reply to: [270627793](https://github.com/dotnet/roslyn/pull/34027#discussion_r270627793) [](ancestors = 270627793)